tools/example-module/example-module.template

```diff
@@ -29,6 +30,9 @@ export const EXAMPLE_LIST = [${exampleList}];
 @NgModule({
   imports: EXAMPLE_MODULES,
   exports: EXAMPLE_MODULES,
+  providers: [
+    { provide: MAT_FORM_FIELD_DEFAULT_OPTIONS, useValue: { appearance: 'fill' } },
```

Comment:
We should add a comment describing why this is necessary. Also won't this cause the docs demo's to look different than what renders in StackBlitz? The code examples will miss this configuration part right?

Reply:
Yes, I'll need to update the StackBlitz template in the docs repo after these changes land.

Reply:
> The code examples will miss this configuration part right?

I'm not sure what you mean.

Reply:
Comment added. PTAL.

Reply:
I meant that the code preview in the docs will render the form-fields with the `fill` appearance, but the actual code will not have anything like `appearance="fill"`. This will/might confuse people as they wonder why the text-fields are displayed with that appearance?

Reply:
Ah, OK I understand now. That's one reason that I wanted to leave those `appearance="fill"` settings on all of the `MatFormField` examples in PR https://github.com/angular/components/pull/18222. So if people are looking at those examples, they will see it.

In the other cases, hopefully it won't be too much confusion, but we can point to the StackBlitz demos to show how `MAT_FORM_FIELD_DEFAULT_OPTIONS` is used.

Reply:
Do you think that I should add the `appearance="fill"` to the input, select, and any other examples in this PR?

Reply:
I don't know what's best here. It's good that we added it explicitly to the form-field examples👍 

For the other examples, I'm unsure. It might not be that big of a deal. I'll leave that up to you and Jeremy.

Reply:
I think it's fine being on the form-field examples and being globally configured elsewhere

Reply:
PR to update our StackBlitz template: https://github.com/angular/material.angular.io/pull/712